test/validate_resnet50/validation.py

```diff
@@ -0,0 +1,263 @@
+
+    for line in output.strip().split("\n"):
+        if line.split()[0] == 'Step':
+            to_parse = True
+            continue
+
+        elif line.startswith('-----'):
+            to_parse = False
+            continue
+
+        if to_parse:
+            total_loss.append(line.split()[-3])
+            top1_acc.append(line.split()[-2])
+            top5_acc.append(line.split()[-1])
+
+    return total_loss, top1_acc, top5_acc
+
+
+def parse_reference_file(filename):
+    to_parse = False
+    total_loss = []
+    top1_acc = []
+    top5_acc = []
+
+    with open(filename) as reference_result:
+        for line in reference_result:
+            if line.split()[0] == 'Step':
+                to_parse = True
+                continue
+
+            elif line.startswith('-----'):
+                to_parse = False
+                continue
+
+            if to_parse:
+                total_loss.append(line.split()[-3])
+                top1_acc.append(line.split()[-2])
+                top5_acc.append(line.split()[-1])
+
+    return total_loss, top1_acc, top5_acc
+
+
+def check_validation_results(norm_dict, metric):
+    test_pass = True
+    for norm in norm_dict:
+        if norm_dict[norm] > 0.1:
+            print(metric + " " + norm +
+                  " is greater than the threshold 0.1, validation failed")
+            test_pass = False
+    return test_pass
+
+
+# Return L1, L2, inf norm of the input arrays
+def calculate_norm_values(result1, result2):
+    l1_norm = np.linalg.norm(
+        (np.array(result1, dtype=np.float) - np.array(result2, dtype=np.float)),
+        1)
+
+    l2_norm = np.linalg.norm(
+        (np.array(result1, dtype=np.float) - np.array(result2, dtype=np.float)),
+        2)
+
+    inf_norm = np.linalg.norm(
+        (np.array(result1, dtype=np.float) - np.array(result2, dtype=np.float)),
+        np.inf)
+    return {"l1_norm": l1_norm, "l2_norm": l2_norm, "inf_norm": inf_norm}
+
+
+def run_validation(data_format, reference_file_name, batch_size):
+    # Apply the patch to make input data loader deterministic for real data validation
+    # Assume the current directory already has the required patch
+    if os.path.isfile('./datasets_make_deterministic.patch'):
+        output, error_output = command_executor(
+            'git apply --check --whitespace=nowarn ' +
+            './datasets_make_deterministic.patch')
+        if error_output:
+            print(
+                "Warning: datasets_make_determinitic.patch is already applied")
+        else:
+            command_executor('git apply --whitespace=nowarn ' +
+                             './datasets_make_deterministic.patch')
+
+    # Run the validation command on NGraph
+    if (data_format == "real_data"):
+        command_to_run = validate_with_real_data_command_NG + str(batch_size)
+    elif (data_format == "synthetic_data"):
+        command_to_run = validate_with_synthetic_data_command_NG + \
+            str(batch_size)
```

Comment:
an `else` can be added that throws an exception to complete out this ladder. perhaps `exit(1)` in keeping with the other parts of this script (eg line 169) in case of fatal errors